store/errors.go

```diff
@@ -107,3 +110,56 @@ func (e InvalidAuthDataError) Error() string {
 	//      (empirically this checks out)
 	return strings.Join(es, "  ")
 }
+
+// SnapActionError conveys errors that were reported on otherwise overall successful snap action (install/refresh) request.
+type SnapActionError struct {
+	// NoResults is set if the there were no results in the response
+	NoResults bool
+	// Refresh errors by snap name.
+	Refresh map[string]error
+	// Install errors by snap name.
+	Install map[string]error
+	// Other errors.
+	Other []error
+}
+
+func (e SnapActionError) Error() string {
+	var es []string
+	if len(e.Refresh) > 0 {
+		es = append(es, "cannot refresh:")
+		for name, e := range e.Refresh {
+			es = append(es, fmt.Sprintf(" - %q: %v", name, e))
+		}
+	}
+	if len(e.Install) > 0 {
+		es = append(es, "cannot install:")
+		for name, e := range e.Install {
+			es = append(es, fmt.Sprintf(" - %q: %v", name, e))
+		}
+	}
+	if len(e.Other) > 0 {
+		es = append(es, "other install/refresh errors:")
+		for _, e := range e.Other {
+			es = append(es, fmt.Sprintf(" - %v", e))
```

Comment:
The format here feels a bit verbose and perhaps unfriendly to read.

Perhaps we can separate this in two cases: if there is only one error, whatever it is, let's spell it out. For example:
```
cannot refresh snap %q: %v"
```
This will probably be the most common case, so it's nice to make it very friendly.

Then, if there are more problems, we can generalize the message:
```
cannot refresh snaps:
snap %q: %s
snap %q: %s
snap %q: %s
```
No need for the initial "-". The first word is already the leading sign.

The first sentence would change according to the errors, but it would be a single message. One of these:
* `cannot install snaps:`
* `cannot refresh snaps:`
* `cannot install or refresh snaps:`